apps/examples/testcase/le_tc/filesystem/fs_main.c

```diff
@@ -2445,6 +2445,111 @@ static void tc_libc_stdio_stdsostream(void)
 }
 
 /**
+* @testcase         tc_libc_stdio_tempnam_tc
+* @brief            Returns a pointer to a string that is a valid filename
+* @scenario         The tempnam() function returns a pointer to a unique temporary filename, or NULL if a unique name cannot be generated.
+* @apicovered       tempnam(), mkstemp(), mktemp()
+* @precondition     NA
+* @postcondition    NA
+*/
+
+static void tc_libc_stdio_tempnam_tc(void)
+{
+	const char filename[] = "tempnam_tc_test";
+	char *ret1 = NULL;
+	char *ret2 = NULL;
+	int ret_check = 0;
+	FILE *fp;
+
+	/* TC 1 with valid prefix */
+
+	ret_check = mount(MOUNT_DEV_DIR, CONFIG_LIBC_TMPDIR, "smartfs", 0, NULL);
+	TC_ASSERT_EQ("mount", ret_check, OK);
+
+	ret1 = tempnam(CONFIG_LIBC_TMPDIR, filename);
+	TC_ASSERT_NEQ_CLEANUP("tempnam", ret1, NULL, goto errout);
+
+	fp = fopen(ret1, "w");
+	TC_ASSERT_NEQ_CLEANUP("fopen", fp, NULL, goto errout1);
+	fclose(fp);
+
+	unlink(ret1);
```

Comment:
I think this can be removed.
If first tc succeed and fopen(ret2) fail, unlink(ret1) is executed also.
And you need to change below errout to errout1.